drivers/flash/soc_flash_ite_it8xxx2.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2020 ITE Corporation. All Rights Reserved.
+ * <jyunlin.chen@ite.com.tw>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <drivers/flash.h>
+#include <errno.h>
+#include <init.h>
+#include <soc.h>
+
+
+#define CONFIG_FLASH_SIZE	DT_REG_SIZE_BY_IDX(DT_NODELABEL(flash0), 0)
+#define FLASH_BLOCK_WRITE_UNIT_SIZE DT_PROP(DT_NODELABEL(flash0),\
+					erase_block_size)
+#define kStatus_Success     0
+
+struct flash_priv {
+	/*
+	 * HACK: flash write protection is managed in software.
```

Comment:
Can you describe this hack more. IS the write_lock never used or something like that?